script.js

```diff
@@ -1,4 +1,102 @@
+var SCISSORS = 'scissors';
+var PAPER = 'paper';
+// Declare an array of the game options
+var gameOptions = [STONE, SCISSORS, PAPER];
+
+
+
+
+// Define a function that randomly generates either scissors, paper or stone
+var randomDraw = function () {
+  console.log('running randomDraw function');
+  
+  // Generate a random decimal between 0 to 3, inclusive of 0 and excluding 3
+  var randomDecimal = Math.random() * 3;
+  // Remove decimal to generate a random integer between 0 to 2, inclusive of 0 and 2
+  var randomInteger = Math.floor(randomDecimal);
+  // Assign string value of the array 'gameOptions' indexed by the random integer to the variable randomOutput
+  var randomOutput = gameOptions[randomInteger];
+
+  console.log('randomOutput');
+  console.log(randomOutput);
+  // Return the randomly chosen option, scissors paper or stone
+  return randomOutput;
+};
+
+
+
+// Define a function that checks if the user input is valid i.e. scissors, paper or stone
+var checkInput = function (str) {
+  // Convert input string characters to lowercase
+  var input = str.toLowerCase();
+  console.log('Running user input validation');
+  if (input != gameOptions[0] && input != gameOptions[1] && input != gameOptions[2]) {
+    return false;
+  }
+  return true;
+}
+
+
+
+var main = function (str) {  
+  // Convert input string characters to lowercase
+  var input = str.toLowerCase();
+  // Input validation: If the user types something other than SPS
+  if (checkInput(input) == false) {
+    console.log('==Invalid input: Request to reenter==');
+    myOutputValue = `Please type either ${gameOptions[0]}, ${gameOptions[1]} or ${gameOptions[2]}` ;
+  } 
+  // If the user input is valid, play the game
+  else {
+    // Computer randomly draws an outcome (Scissors, Paper or Stone) that is stored in a new variable
+    var comOutput = randomDraw();
+    // Update the number of attempts
+    attemptsCount += 1; 
+
+    // If it is a draw
+    if (input == comOutput) {
+      //Update the number of draws
+      drawCount += 1;
+      console.log('** Draw **');
+      myOutputValue = `You chose ${input} and the computer chose ${comOutput}.<br>It is a draw :-)`;
+    } 
+    // If the user wins 
+    else if ( (input == SCISSORS && comOutput == PAPER) || (input == PAPER && comOutput == STONE) || (input == STONE && comOutput == SCISSORS) ){
+      // Update the number of user wins
+      winCount_user += 1;
+      console.log('** User winning **');
+      myOutputValue = `You chose ${input} and the computer chose ${comOutput}. <br>You win :) `;
+    } 
+    // If the computer wins
+    else if ((comOutput == SCISSORS && input == PAPER) || (comOutput == PAPER && input == STONE) || (comOutput == STONE && input == SCISSORS)) {
+      // Update the number of computer wins
+      winCount_com += 1;
+      console.log('** Computer winning **');
+      myOutputValue = `You chose ${input} and the computer chose ${comOutput}. <br>You lose. <br>Try again :D`;
+    }
+
+  }
```

Comment:
We can simplify the else if logic with multiple return statements. 

See this [example ](https://basics.rocketacademy.co/8-managing-state-and-input-validation/8.3-input-validation#simplify-logic-with-multiple-return-statements-in-single-function)and compare it to the one above